### PR TITLE
fix(p2p): adjust backoff seconds to increase reconnect retries close to 24 hours

### PR DESCRIPTION
**Pull request by andynog**
close: #3519 

Adjust `reconnectBackOffBaseSeconds` to increase reconnect retries to up 1 day (~24 hours).

The new value can be validated here: https://go.dev/play/p/k8F5rS-i24p, which will show that the total time is increased to almost 24 hours. 

Initial reconnecting time: 2m8.493s
Total reconnecting time. : 23h55m56.249s

The `reconnectBackOffBaseSeconds` is increased by a bit over 10% (from 3.0 to 3.4 seconds) so this would not affect reconnection retries too much.

#### PR checklist

- [ ] ~~Tests written/updated~~
- [x] Changelog entry added in `.changelog` (we use [unclog](https://github.com/informalsystems/unclog) to manage our changelog)
- [x] Updated relevant documentation (`docs/` or `spec/`) and code comments

### CODE REVIEW

**Review by melekes — APPROVED** [review]
👍

**Comment by cason** [comment]
Hum, I think the problem is not on the base of the backoff, but the fact that the second loop starts from `0`:

https://go.dev/play/p/mm3Hb13m5Cs

File: p2p/switch.go [hunk 1]
```diff
@@ -387,7 +387,8 @@ func (sw *Switch) stopAndRemovePeer(p Peer, reason any) {
 }
 
 // reconnectToPeer tries to reconnect to the addr, first repeatedly
-// with a fixed interval, then with exponential backoff.
+// with a fixed interval (approximately 2 minutes), then with
+// exponential backoff (approximately close to 24 hours).
```
Comment (cason) [inline p2p/switch.go:391]: The problem here for me is the second four (lines 424/426) starts from 0. Anything with power 0 == 1.

**Review by cason — CHANGES_REQUESTED** [review]
Thank you for addressing this, but I think the solution goes in another direction (see my comments).

**Comment by andynog** [comment]
Thanks @cason, I've made the suggested changes.

File: p2p/switch.go [hunk 2]
```diff
@@ -423,7 +425,7 @@ func (sw *Switch) reconnectToPeer(addr *na.NetAddr) {
 
 	sw.Logger.Error("Failed to reconnect to peer. Beginning exponential backoff",
 		"addr", addr, "elapsed", time.Since(start))
-	for i := 0; i < reconnectBackOffAttempts; i++ {
+	for i := 1; i < reconnectBackOffAttempts; i++ {
```
Comment (cason) [inline p2p/switch.go:428]:
```suggestion
	for i := 1; i <= reconnectBackOffAttempts; i++ {
```

Comment (cason) [reply p2p/switch.go:428]: Otherwise it takes 8h.

**Review by cason — APPROVED** [review]
Thank for this. There is a `<` that must become `<=`.

**Comment by andynog** [comment]
Thanks @cason, should we also back port this to `v0.38.x` and `v0.37.x` seems the logic is the same.

**Comment by cason** [comment]
> should we also back port this to v0.38.x and v0.37.x seems the logic is the same.

I think we should.

**Pull request merged**